examples/tutorials/rigid_object_primitive_tutorial.py

```diff
@@ -0,0 +1,198 @@
+output_path = os.path.join(dir_path, "rigid_object_primitive_tutorial_output/")
+
+
+def make_video_cv2(
+    observations, camera_res=[540, 720], prefix="", open_vid=True, multi_obs=False
+):
+    videodims = (camera_res[1], camera_res[0])
+    fourcc = cv2.VideoWriter_fourcc("m", "p", "4", "v")
+    video = cv2.VideoWriter(output_path + prefix + ".mp4", fourcc, 60, videodims)
+    thumb_size = (int(videodims[0] / 5), int(videodims[1] / 5))
+    outline_frame = np.ones((thumb_size[1] + 2, thumb_size[0] + 2, 3), np.uint8) * 150
+
+    if multi_obs:
+        for ob in observations:
+            embed_image_data_list = [
+                (ob["rgba_camera_1stperson"], False),
+                (ob["depth_camera_1stperson"], True),
+            ]
+
+            res_image = tutFuncs.build_multi_obs_image(
+                ob["rgba_camera_3rdperson"],
+                embed_image_data_list,
+                thumb_size,
+                outline_frame,
+            )
+            # write the desired image to video
+            video.write(res_image)
+    else:
+        for ob in observations:
+            res_image = ob["rgba_camera_1stperson"][..., 0:3][..., ::-1]
+
+            # write the desired image to video
+            video.write(res_image)
+    video.release()
+    if open_vid:
+        os.system("open " + output_path + prefix + ".mp4")
+
+
+def make_tutorial_configuration(camera_resolution=[540, 720]):
+    # sensor configurations
+    # Note: all sensors must have the same resolution
+    # setup 2 rgb sensors for 1st and 3rd person views
+    sensors_config_dict = {
+        "rgba_camera_1stperson": {
+            "sensor_type": habitat_sim.SensorType.COLOR,
+            "resolution": camera_resolution,
+            "position": [0.0, 0.6, 0.0],
+            "orientation": [0.0, 0.0, 0.0],
+        },
+        "depth_camera_1stperson": {
+            "sensor_type": habitat_sim.SensorType.DEPTH,
+            "resolution": camera_resolution,
+            "position": [0.0, 0.6, 0.0],
+            "orientation": [0.0, 0.0, 0.0],
+        },
+        "rgba_camera_3rdperson": {
+            "sensor_type": habitat_sim.SensorType.COLOR,
+            "resolution": camera_resolution,
+            "position": [0.0, 1.0, 0.3],
+            "orientation": [-45, 0.0, 0.0],
+        },
+    }
+
+    return tutFuncs.make_configuration("apartment_1.glb", True, sensors_config_dict)
+
+
+def simulate(sim, dt=1.0, get_frames=True):
+    # simulate dt seconds at 60Hz to the nearest fixed timestep
+    print("Simulating " + str(dt) + " world seconds.")
+    observations = []
+    start_time = sim.get_world_time()
+    while sim.get_world_time() < start_time + dt:
+        sim.step_physics(1.0 / 60.0)
+        if get_frames:
+            observations.append(sim.get_sensor_observations())
+
+    return observations
+
+
+# [/setup]
+
+# This is wrapped such that it can be added to a unit test
+def main(make_video=True, show_video=True):
```

Comment:
Explicitly wrapping a main function like this will actually make it harder to step through a notebook.

Reply:
What would be a better alternative?